src/storage/txn/commands/flashback_to_version_read_phase.rs

```diff
@@ -81,24 +81,42 @@ impl<S: Snapshot> ReadCommand<S> for FlashbackToVersionReadPhase {
             self.tag().get_str(),
             (key_locks.len() + key_old_writes.len()) as f64,
         );
-
+        // Check next key firstly.
+        let next_lock_key = if has_remain_locks {
+            key_locks.pop().map(|(key, _)| key)
+        } else {
+            None
+        };
+        let next_write_key = if has_remain_writes && !key_old_writes.is_empty() {
```

Comment:
How about `using` match instead?
```suggestion
                let next_write_key = match (has_remain_writes, key_old_writes.is_empty()) {
            (true, false) => key_old_writes.pop().map(|(key, _)| key),
            (true, true) => self.next_write_key,
            (..) => None,
        };
```

Reply:
get it